streams/src/main/java/org/apache/kafka/streams/KafkaStreaming.java

```diff
@@ -68,12 +71,12 @@
     private static final int STOPPED = 2;
     private int state = CREATED;
 
-    private final StreamThread[] threads;
+    private final List<StreamThread> threads;
```

Comment:
I was originally thinking arrays should be avoided in general, also suggested by someone from OS reviews:

http://eclipsesource.com/blogs/2014/04/11/3-good-reasons-to-avoid-arrays-in-java-interfaces/

But thinking of this a bit more now I feel this is fine in our case here. Will change it back.